### DISCUSSION TITLE
用 JSON 格式是不是 emoji 就显示不出来了？

### DISCUSSION THREAD

**Discussion by Hellager** [Zhengqbbb/cz-git#76] [Q&A] [answered]
普通的 .commitlintrc.js 我看了这个 issue 配出来没问题 https://github.com/Zhengqbbb/cz-git/issues/49
因为我的 package.json 里有的有 type: module 字段，所以 .commitlintrc.js 在这个情况下不能直接用，用  json 格式的时候没问题，所以打算试试配 JSON
看文档有说要有 prompt key, 把文档里 JSON 的例子照着 emoji 模板敲了几个 emoji 进去，类似这样
```js
// .czrc | package.json | .commitlintrc(need "prompt" key)

// .commitlintrc.json
{ "value": "feat", "name": "feat:     ✨  A new feature", "emoji": ":sparkles:" }

// package.json
  "config": {
    "commitizen": {
      "path": "node_modules/cz-git"
    }
  },
```
然后发现 emoji 没显示出来，“useEmoji” 设置为 true 后实际的提交信息里也没有 emoji
```shell
chore: test emoji
```
JSON 格式这 emoji 该怎么配？

**Comment by Zhengqbbb** [comment, marked as answer]
> `.commitlintrc`(need "prompt" key)
```json
{
  "prompt": {
    "types": [
      { "value": "feat", "name": "feat:     A new feature", "emoji": ":sparkles:" }
    ],
    "useEmoji": true
  }
}
```

<img width="935" alt="image" src="https://user-images.githubusercontent.com/40693636/201013554-5b101db1-d9b5-4ceb-8106-54c3b24e6b08.png">

But if u project is esm project ` type: module`. u can use `.cjs` configure like `commitlint.config.cjs`| `.commitlintrc.cjs` | `cz.config.cjs`

```js
/** @type {import('cz-git').UserConfig} */
module.exports = {
  prompt: {
    types: [
      { value: "feat", name: "feat:     A new feature", emoji: ":sparkles:" },
    ],
    useEmoji: true
  },
};
```

![image](https://user-images.githubusercontent.com/40693636/201014429-a5b437c3-fdad-479f-9ec4-6848efdd81e0.png)

**Reply by Hellager** [reply]
That's it! Great thanks for your help.
